x/rollapp/keeper/state_info.go

```diff
@@ -85,3 +93,50 @@ func (k Keeper) GetAllStateInfo(ctx sdk.Context) (list []types.StateInfo) {
 
 	return
 }
+
+// DeleteStateInfoUntilTimestamp deletes all stateInfo until the given timestamp
+func (k Keeper) DeleteStateInfoUntilTimestamp(ctx sdk.Context, endTimestampExcl time.Time) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.StateInfoKeyPrefix))
+	storeTS := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.TimestampedStateInfoKeyPrefix))
+
+	// get all latest stateInfo and latest finalized stateInfo and add them to a map
+	latestIndexes := append(k.GetAllLatestStateInfoIndex(ctx), k.GetAllLatestFinalizedStateIndex(ctx)...)
```

Comment:
this comment should explain why/reasoning, not just tell the code in English
especially since this is an important point

Reply:
also, isn't it true that if latest is at prunable age .e.g. 3 weeks, then it is also for sure finalized?

So why need both here?